org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileManager.java

```diff
@@ -1,15 +1,65 @@
 package org.code.javabuilder;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+
+/** Manages the in-memory list of user project files. */
 public class UserProjectFileManager implements ProjectFileManager {
+  private final String baseUrl;
+  private final String[] fileNames;
+  private final ArrayList<ProjectFile> fileList;
 
-  public UserProjectFileManager(String baseUrl, String[] fileNames) {}
+  public UserProjectFileManager(String baseUrl, String[] fileNames) {
+    this.baseUrl = baseUrl;
+    this.fileNames = fileNames;
+    this.fileList = new ArrayList<>();
+  }
 
+  /**
+   * Asynchronously loads all user project files.
+   *
+   * @throws UserFacingException If the load times out or fails.
+   */
   public void loadFiles() throws UserFacingException {
-    throw new UserFacingException("We couldn't load your files");
+    ExecutorService executor = Executors.newCachedThreadPool();
+    List<Callable<Boolean>> callables = new ArrayList<>();
```

Comment:
(very minor) Consider renaming `callables` to something that indicates that it's the list of requests that will fetch files (e.g. `fileLoaders`, `loadRequests`, etc.)

Reply:
✔️